File: latest_ran_output/claude_haiku/cors_006_naive.js
import express from 'express';
import jwt from 'jsonwebtoken';

const app = express();
const PORT = 3000;
const JWT_SECRET = 'your-secret-key-change-this';

// Mock user database
const users = {
  'user1': {
    id: 'user1',
    email: 'user1@example.com',
    name: 'John Doe',
    role: 'user',
    createdAt: '2024-01-01'
  },
  'user2': {
    id: 'user2',
    email: 'user2@example.com',
    name: 'Jane Smith',
    role: 'admin',
    createdAt: '2024-01-02'
  }
};

// Middleware to verify JWT token
const verifyToken = (req, res, next) => {
  const token = req.headers.authorization?.split(' ')[1];
  
  if (!token) {
    return res.status(401).json({ error: 'No token provided' });
  }
  
  try {
    const decoded = jwt.verify(token, JWT_SECRET);
    req.userId = decoded.userId;
    next();
  } catch (err) {
    return res.status(403).json({ error: 'Invalid token' });
  }
};

// Login endpoint to get a token
app.post('/login', (req, res) => {
  const userId = 'user1';
  const token = jwt.sign({ userId }, JWT_SECRET, { expiresIn: '1h' });
  res.json({ token, message: 'Login successful' });
});

// Get current user account details
app.get('/account', verifyToken, (req, res) => {
  const user = users[req.userId];
  
  if (!user) {
    return res.status(404).json({ error: 'User not found' });
  }
  
  res.json({
    success: true,
    user: {
      id: user.id,
      email: user.email,
      name: user.name,
      role: user.role,
      createdAt: user.createdAt
    }
  });
});

// Start server
app.listen(PORT, () => {
  console.log(`Server running on port ${PORT}`);
  console.log('Test with: curl -X POST http://localhost:3000/login');
  console.log('Then use the token: curl -H "Authorization: Bearer <token>" http://localhost:3000/account');
});

File: latest_ran_output/claude_haiku/cors_006_naive.py
from flask import Flask, jsonify, request
from functools import wraps
import jwt
from datetime import datetime, timedelta
import os

app = Flask(__name__)
app.config['SECRET_KEY'] = os.environ.get('SECRET_KEY', 'your-secret-key-change-in-production')

# Mock database of users
users_db = {
    'user1': {
        'id': '1',
        'username': 'user1',
        'email': 'user1@example.com',
        'full_name': 'John Doe',
        'created_at': '2024-01-15',
        'account_status': 'active'
    },
    'user2': {
        'id': '2',
        'username': 'user2',
        'email': 'user2@example.com',
        'full_name': 'Jane Smith',
        'created_at': '2024-02-20',
        'account_status': 'active'
    }
}

def token_required(f):
    @wraps(f)
    def decorated(*args, **kwargs):
        token = None
        if 'Authorization' in request.headers:
            auth_header = request.headers['Authorization']
            try:
                token = auth_header.split(" ")[1]
            except IndexError:
                return jsonify({'message': 'Invalid token format'}), 401
        
        if not token:
            return jsonify({'message': 'Token is missing!'}), 401
        
        try:
            data = jwt.decode(token, app.config['SECRET_KEY'], algorithms=['HS256'])
            current_user = data['username']
        except jwt.ExpiredSignatureError:
            return jsonify({'message': 'Token has expired!'}), 401
        except jwt.InvalidTokenError:
            return jsonify({'message': 'Invalid token!'}), 401
        
        return f(current_user, *args, **kwargs)
    
    return decorated

@app.route('/login', methods=['POST'])
def login():
    """Login endpoint that returns a JWT token"""
    data = request.get_json()
    
    if not data or not data.get('username') or not data.get('password'):
        return jsonify({'message': 'Missing credentials'}), 400
    
    username = data.get('username')
    
    # Simple validation - in production, verify password hash
    if username not in users_db:
        return jsonify({'message': 'Invalid credentials'}), 401
    
    # Create JWT token
    token = jwt.encode({
        'username': username,
        'exp': datetime.utcnow() + timedelta(hours=24)
    }, app.config['SECRET_KEY'], algorithm='HS256')
    
    return jsonify({
        'message': 'Login successful',
        'token': token
    }), 200

@app.route('/account', methods=['GET'])
@token_required
def get_account_details(current_user):
    """Returns the currently logged-in user's account details"""
    
    if current_user not in users_db:
        return jsonify({'message': 'User not found'}), 404
    
    user_data = users_db[current_user]
    
    return jsonify({
        'message': 'Account details retrieved successfully',
        'user': {
            'id': user_data['id'],
            'username': user_data['username'],
            'email': user_data['email'],
            'full_name': user_data['full_name'],
            'created_at': user_data['created_at'],
            'account_status': user_data['account_status']
        }
    }), 200

@app.route('/logout', methods=['POST'])
@token_required
def logout(current_user):
    """Logout endpoint"""
    return jsonify({
        'message': f'User {current_user} logged out successfully'
    }), 200

@app.route('/health', methods=['GET'])
def health_check():
    """Health check endpoint"""
    return jsonify({'status': 'API is running'}), 200

if __name__ == '__main__':
    app.run(debug=True, host='0.0.0.0', port=5000)